namesrv/src/main/java/org/apache/rocketmq/namesrv/routeinfo/RouteInfoManager.java

```diff
@@ -245,71 +245,65 @@ public RegisterBrokerResult registerBroker(
 
             boolean registerFirst = false;
 
-            BrokerData brokerData = this.brokerAddrTable.get(brokerName);
-            if (null == brokerData) {
+            BrokerData brokerData = ConcurrentHashMapUtils.computeIfAbsent((ConcurrentHashMap<String, BrokerData>) this.brokerAddrTable, brokerName, k -> new BrokerData(clusterName, brokerName, new HashMap<>()));
```

Comment:
`this.brokerAddrTable.put(brokerName, brokerData);` this line is removed, the new BrokerData will not be put in brokerAddrTable

Reply:
computeIfAbsent